src/randomx.cpp

```diff
@@ -363,12 +363,12 @@ extern "C" {
 		machine->getFinalResult(output, RANDOMX_HASH_SIZE);
 	}
 
-	void randomx_calculate_hash_first(randomx_vm* machine, uint64_t (&tempHash)[8], const void* input, size_t inputSize) {
+	void randomx_calculate_hash_first(randomx_vm* machine, uint64_t *tempHash, const void* input, size_t inputSize) {
```

Comment:
For typechecking purposes you could still typedef this param to `uint64_t foo[8]` and use the type in the parameter lists.

Reply:
Yeah, `uint64_t tempHash[8]` could be more readable in the API and it decays into a pointer anyways.

Reply:
And that would have prevented that sizeof() bug too.

Reply:
Using the "first" and "next" hash function in randomx.cpp, the miner (mine which based on this repo, not xmirg) receives "Low difficulty share" from pool.

Reply:
You have to be careful using these new functions, check how benchmark.cpp handles them and where it updates the nonce.

Reply:
My miner works fine on this workflow 
```
	uint64_t tempHash[8];
	while (nonce < noncesCount) {
		nonce = atomicNonce.fetch_add(1);
		store32(noncePtr, nonce);
	        randomx_calculate_hash_first(vm, tempHash, blockTemplate, sizeof(blockTemplate));
		randomx_calculate_hash_next(vm, tempHash, blockTemplate, sizeof(blockTemplate), &hash);
		result.xorWith(hash);
	}
```

But when the `randomx_calculate_hash_first` func is out of while loop, the miner gets the wrong result.
```
	uint64_t tempHash[8];

	store32(noncePtr, nonce);
	randomx_calculate_hash_first(vm, tempHash, blockTemplate, sizeof(blockTemplate));

	while (nonce < noncesCount) {
		nonce = atomicNonce.fetch_add(1);
		store32(noncePtr, nonce);
		randomx_calculate_hash_next(vm, tempHash, blockTemplate, sizeof(blockTemplate), &hash);
		result.xorWith(hash);
	}
```